aikau/src/main/resources/alfresco/renderers/InlineEditProperty.js

```diff
@@ -207,33 +260,25 @@ define(["dojo/_base/declare",
       },
 
       /**
-       * References the widget used for editing. Created by calling the 
-       * [getFormWidget]{@link module:alfresco/renderers/InlineEditProperty#getFormWidget}
-       * for the first time.
-       * 
-       * @instance
-       * @type {object}
-       * @default null
-       */
-      formWidget: null,
-
-      /**
-       * <p>In certain circimstances it may be necessary to submit additional data along with that
-       * provided by the main edit control. This configuration property should take the form:</p>
-       * <p><pre>hiddenDataRules: [
-       *   {
-       *     name: "customProperties",
-       *     rulePassValue: "hiddenData",
-       *     ruleFailValue: "",
-       *     is: ["includeHiddenData"]
-       *   }
-       * ]</pre></p>
+       * Extends the [inherited function]{@link module:alfresco/renderers/Property#postCreate} to
+       * check the [permissionProperty]{@link module:alfresco/renderers/InlineEditProperty#permissionProperty}
+       * to determine whether or not the current user actually has permission to edit the current item. If
+       * the user does not have permission then then edit controls will be hidden (and keyboard shortcuts suppressed).
        *
        * @instance
-       * @type {array}
-       * @default null
        */
-      hiddenDataRules: null,
+      postCreate: function alfresco_renderers_InlineEditProperty__postCreate() {
+         this.inherited(arguments);
+         if (this.permissionProperty)
+         {
+            var hasEditPermission = lang.getObject(this.permissionProperty, false, this.currentItem);
+            if (!hasEditPermission)
+            {
+               domStyle.set(this.editIconNode, "display", "none");
```

Comment:
Can we instead use something like ...

```
domClass.add(this.domNode, "alfresco-renderers-InlineEditProperty--disabled");
```

... instead? Semantically, it's much better and gives greater scope for further changes just by tweaking the CSS, especially when it comes to external customisation. Direct style manipulation is also impossible to override with a stylesheet unless people use `!important` which is obviously __ABadThing<sup>TM</sup>__.